### ISSUE TITLE
Referencing assets (fonts, images) in SCSS/CSS files?

### ISSUE THREAD

**Issue by nathobson** [ravorona/sage#12]
Thanks for your work on this project, super interesting.

One issue I've hit is referencing static assets such as fonts or images. I've tried:

- relative i.e. `./font-name.woff2` or `../fonts/font-name.woff2`
- Using the Vite aliases i.e. `@/fonts/font-name.woff2`

But these always return paths relative to the root of the whole site i.e. `site.test/fonts/font-name.woff2` or `site.test/resources/fonts/font-name.woff2`.

The Vite docs say to use a `public` directory but this gets wiped out by a build as Sage also expects a `public` directory for compiled assets, so I can see the Vite `out` directory is set to `public` and Sage references `public` for enqueuing the  compiled CSS/JS files.

Any pointers here?

**Comment by nathobson** [comment]
OK, so after a small break and coming back to this, it appears that setting `base: './'` within `vite.config.ts` and referencing using aliases `@/fonts/font-name.woff2` works for `build` but in `dev` it's missing the port in the URL. The CSS file is trying to reference `site.test/fonts/font-name.woff2` rather than `site.test:3000/fonts/font-name.woff2`.

**Comment by nathobson** [comment]
I've now managed to get this working. The issue is around Vite not supporting the dev server having a different URL, see: https://github.com/vitejs/vite/issues/2196.

I've added `base: './',` near the top of the config just after `publicDir` to process the URLs during build (relative to the theme) and then I've modified the dev server config to include `origin: devServerConfig.HMR_ENTRYPOINT`, as follows:

```
  if (dev) {
    config.server = {
      host: '0.0.0.0',
      origin: devServerConfig.HMR_ENTRYPOINT,
      strictPort: true,
      fs: {
        strict: true,
        allow: ['node_modules', assets.base]
      }
    }
```
This adds the correct URLs during dev.
